### DISCUSSION TITLE
Allow "select all" / "select none" for "Filter by name" transform

### DISCUSSION THREAD

**Discussion by slhck** [grafana/grafana#58904] [Dashboards, Panels & Visualizations]
This is a feature request for the "Filter by name" transform.

When querying results from an Elasticsearch index, I get a large number of fields returned. I do not want to filter the results in the Elasticsearch query, but rather with the transformation.

This is a tedious process given the long list of fields:

<img width="816" alt="image" src="https://user-images.githubusercontent.com/582444/202458092-de706f7c-fb4e-4255-8bd2-983f86275318.png">

There are several UX issues here:

* I can only individually click and select/deselect values, not select all or deselect all
* The UI responsiveness here is very slow, as each click causes a rerender of the panel
* Finding an invidual field is hard. While they are sorted, since they take up a different amount of vertical space per item, it's almost impossible to visually identify a single item.

I'd therefore suggest to:

* Include a "select all" and "deselect all" option for making coarse selections.
* Align the items in a list (or columns) to make it easier to find them

Sloppy mockup here:

![image](https://user-images.githubusercontent.com/582444/202461364-1c7e0bd7-3610-47aa-b518-404cb280141a.png)

**Comment by rossapi** [comment]
totally agree

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
